test/Solnet.Programs.Test/TokenProgramTest.cs

```diff
@@ -792,5 +919,501 @@ public void TestThawAccountMultiSignature()
+            Assert.IsTrue(decodedInstructions[0].Values.TryGetValue("Account", out object source));
+            Assert.IsTrue(decodedInstructions[0].Values.TryGetValue("Mint", out object mint));
+            Assert.IsTrue(decodedInstructions[0].Values.TryGetValue("Authority", out object authority));
+            Assert.IsTrue(decodedInstructions[0].Values.TryGetValue("Amount", out object amount));
+            Assert.IsTrue(decodedInstructions[0].Values.TryGetValue("Decimals", out object decimals));
+            Assert.IsTrue(decodedInstructions[0].Values.TryGetValue("Signer 1", out object signer1));
+            Assert.IsTrue(decodedInstructions[0].Values.TryGetValue("Signer 2", out object signer2));
+            Assert.IsTrue(decodedInstructions[0].Values.TryGetValue("Signer 3", out object signer3));
+            Assert.AreEqual("6ULjh7zKSsbiMqNzpCKMgYCUDFZ9Thxy59SGE48x22AF", (PublicKey)source);
+            Assert.AreEqual("HAbjCwXvJLRYPwsfLftT52iPYbHeGPyhPi3QCWiY2CTq", (PublicKey)mint);
+            Assert.AreEqual("7L1UA4SsaH3AonYX8mHXbLaLR3fiJY71S8zSZQ57WXv8", (PublicKey)authority);
+            Assert.AreEqual("6yg3tZM1szHj752RDxQ1GxwvkzR3GyuvAcH498ew1t2T", (PublicKey)signer1);
+            Assert.AreEqual("88SzfLipgVTvi8hQwYfq21DgQFcABx6yAwgJH5shfqVZ", (PublicKey)signer2);
+            Assert.AreEqual("5Xcw7EQb6msgpVdGB8Hf8kpCqVyacTChgFBUphpuUeBo", (PublicKey)signer3);
+            Assert.AreEqual(1999020000UL, (ulong)amount);
+            Assert.AreEqual(10, (byte)decimals);
+            
+            Assert.AreEqual("Close Account", decodedInstructions[1].InstructionName);
+            Assert.AreEqual("Token Program", decodedInstructions[1].ProgramName);
+            Assert.AreEqual("TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA", decodedInstructions[1].PublicKey);
+            Assert.AreEqual(0, decodedInstructions[1].InnerInstructions.Count);
+            Assert.IsTrue(decodedInstructions[1].Values.TryGetValue("Account", out source));
+            Assert.IsTrue(decodedInstructions[1].Values.TryGetValue("Destination", out object destination));
+            Assert.IsTrue(decodedInstructions[1].Values.TryGetValue("Authority", out authority));
+            Assert.IsTrue(decodedInstructions[1].Values.TryGetValue("Signer 1", out signer1));
+            Assert.IsTrue(decodedInstructions[1].Values.TryGetValue("Signer 2", out signer2));
+            Assert.IsTrue(decodedInstructions[1].Values.TryGetValue("Signer 3", out signer3));
+            Assert.AreEqual("6ULjh7zKSsbiMqNzpCKMgYCUDFZ9Thxy59SGE48x22AF", (PublicKey)source);
+            Assert.AreEqual("5omQJtDUHA3gMFdHEQg1zZSvcBUVzey5WaKWYRmqF1Vj", (PublicKey)destination);
+            Assert.AreEqual("6yg3tZM1szHj752RDxQ1GxwvkzR3GyuvAcH498ew1t2T", (PublicKey)signer1);
+            Assert.AreEqual("88SzfLipgVTvi8hQwYfq21DgQFcABx6yAwgJH5shfqVZ", (PublicKey)signer2);
+            Assert.AreEqual("5Xcw7EQb6msgpVdGB8Hf8kpCqVyacTChgFBUphpuUeBo", (PublicKey)signer3);

```

Comment:
Question: aren't these casts to `PublicKey` redundant? IIRC the `AreEqual` method accepts `object` type.

Reply:
Yeah I think it would've worked now that there's the `ToString()` previously it wouldn't work as well iirc